bson/src/main/org/bson/Document.java

```diff
@@ -158,6 +160,67 @@ public Document append(final String key, final Object value) {
         return value == null ? defaultValue : (T) value;
     }
 
+    /**
+     * Gets the value of the given list of keys, casting it to the given {@code Class<T>}.  This is useful to avoid having casts in
+     * client code, though the effect is the same.  So to get the embedded value of a key list that is of type String, you would write
+     * {@code String name = doc.getEmbedded(List.of("book", "title"), String.class)} instead of
+     * {@code String name = (String) doc.get("book").get("title") }.
+     *
+     * @param keys  the list of keys
+     * @param clazz the non-null class to cast the value to
+     * @param <T>   the type of the class
+     * @return the value of the given embedded key, or null if the instance does not contain this embedded key.
+     * @throws ClassCastException if the value of the given embedded key is not of type T
```

Comment:
All new methods should include a `@since` annotation.

Reply:
Done